### DISCUSSION TITLE
Why does Switch use a button under the hood and not a checkbox?

### DISCUSSION THREAD

**Discussion by Aaronius** [tailwindlabs/headlessui#928] [General]
Thanks for the great library! Why does Switch use a button under the hood and not a checkbox? I'm more curious than anything, as a switch seems to just be a different way to present a checkbox.

**Comment by dospunk** [comment]
The button uses `role="switch"`, which is explained [here](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/switch_role) as being more specific than a checkbox because instead of representing "checked" and "unchecked" states, it represents "on" and "off" states. I'm not a maintainer so I can't speak to their exact reasoning, but I imagine this is it.

**Reply by Aaronius** [reply]
Interesting! Thanks!
